packages/selinux-policy/systems.cil

```diff
@@ -21,5 +21,7 @@
 (classmapping systems use (key (all)))
 (classmapping systems use (lockdown (all)))
 (classmapping systems use (perf_event (all)))
+(classmapping systems use (io_uring (all)))
```

Comment:
`io_uring (cmd)` seems similar to `file (ioctl)` and could also go into the `files (load)` mapping. 

`io_uring (sqpoll)` and `io_uring (override_creds)` are trickier since they involve credentials.

`io_uring (sqpoll)` appears to be "reflexive" in that `selinux_uring_sqpoll()` only ever checks whether the current sid has permission for the current sid.

There are blanket rules in `rules.cil` covering these two "self" cases:
```
(allow all_s self (processes (interact)))
(allow all_s self (processes (transform)))
```

Of the two, `processes (interact)` is probably the best fit, since "transform" is more for SELinux-related label changes.

Since `io_uring (override_creds)` allows one domain to impersonate another, that one should go in the `processes (transform)` map.

Reply:
I am trying to wrap my head around these, so if I understand correctly we want to move these out of the `systems use` one and into more fine grained mappings. If I understand correctly we would want yo do something like the following?

> `io_uring (cmd)` seems similar to `file (ioctl)` and could also go into the `files (load)` mapping.

Add something like

```
(classmapping files load (io_uring (cmd)))
```

to either [files.cil - line 60](https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/selinux-policy/files.cil#L60) or a new file concerned with io_uring rules.

> `io_uring (sqpoll)` and `io_uring (override_creds)` are trickier since they involve credentials.
> 
> `io_uring (sqpoll)` appears to be "reflexive" in that `selinux_uring_sqpoll()` only ever checks whether the current sid has permission for the current sid.
> 
> There are blanket rules in `rules.cil` covering these two "self" cases:
> 
> ```
> (allow all_s self (processes (interact)))
> (allow all_s self (processes (transform)))
> ```
> 
> Of the two, `processes (interact)` is probably the best fit, since "transform" is more for SELinux-related label changes.
> 
> Since `io_uring (override_creds)` allows one domain to impersonate another, that one should go in the `processes (transform)` map.


Probably similar, but something like
```
(classmapping processes interact (io_uring (sqpoll)))
(classmapping processes transform (io_uring (override_creds)))
```

to either a suitable place in the current files or a new file for io_uring mappings.

Or am I completely misunderstanding what you are proposing?

Reply:
Yup, that's what I had in mind - expanding the current maps in `files.cil` and `processes.cil` the way you've indicated here.

I might structure the changes to `files.cil` a bit differently for consistency with the rest of the mappings, e.g.:
```
(classmapping files load load_io_uring)
(classpermission load_io_uring)
(classpermissionset load_io_uring (
  io_uring (cmd))
```

It's more boilerplate than the implicit syntax, but I find it helpful to be able to see at a glance where some of the classes show up. E.g. "filesystem" is only in some of the maps, and "fd" is only in load.